statsd/src/main/java/com/avast/metrics/statsd/StatsDMetricsMonitor.java

```diff
@@ -244,7 +244,7 @@ protected String constructMetricName(String finalName) {
 
     protected String constructMetricName(Optional<String> finalName) {
         List<String> copy = new ArrayList<>(names);
-        finalName.ifPresent(copy::add);
+        finalName.map(this::sanitizeNames).ifPresent(copy::add);
```

Comment:
Do we want to check also max. length? Max. filename length (not full path) is 256 characters in Linux, Graphite stores the data to directory structure, I don't unfortunately know details.

Reply:
Datadog has a limit of 200 characters. It might make sense to check it as well (what should we do if it's longer?), but I don't want to solve it within the same PR: